src/register/pmpaddrx.rs

```diff
@@ -59,7 +59,7 @@ impl PmpAddr {
         // verify size is not too big
         if (encoded_size > usize::MAX as Size) ||
             // check size is a power of 2
-            (size == (size & !(size-1))) ||
+            (size != (size & !(size-1))) ||
```

Comment:
Wouldn't it be easier to just check `((size & (size-1)) != 0)`? It effectively does the same thing, but I find the pattern easier to decode :rofl:

Reply:
Hm, I don't quite follow.  `((size & (size-1)) = size` iff size is a power of 2, so wouldn't `((size & (size-1)) != 0)` be true when size is a power of two?

Reply:
Consider size  = 4 (0b100)

size - 1 = 3 (0b011)

size & (size - 1)  = (0b100 & 0b011) = 0b000

So, size & (size - 1) returns 0 when size is a power of 2.

Reply:
Oh, I didn't notice you dropped the `!` before (size-1)!

Reply:
changed.